app/src/main/java/ch/epfl/sdp/musiconnect/MapsActivity.java

```diff
@@ -305,7 +373,129 @@ public void onInfoWindowClick(Marker marker) {
         }
     }
 
+    protected boolean checkConnection() {
+        ConnectivityManager connectivityManager = (ConnectivityManager) getSystemService(Context.CONNECTIVITY_SERVICE);
+        if (connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_MOBILE).getState() == NetworkInfo.State.CONNECTED ||
+                connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_WIFI).getState() == NetworkInfo.State.CONNECTED) {
+            //we are connected to a network
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    protected boolean checkLocationServices(){
+        LocationManager lm = (LocationManager)MapsActivity.this.getSystemService(Context.LOCATION_SERVICE);
+        boolean gps_enabled = false;
+        boolean network_enabled = false;
 
+        try {
+            gps_enabled = lm.isProviderEnabled(LocationManager.GPS_PROVIDER);
+        } catch(Exception ex) {}
+
+        try {
+            network_enabled = lm.isProviderEnabled(LocationManager.NETWORK_PROVIDER);
+        } catch(Exception ex) {}
+
+        if(!gps_enabled && !network_enabled) {
+            return false;
+        } else{
+            return true;
+        }
+
+    }
+
+    //creates warning message when something goes wrong; type 1 is a simple message at the bottom of the screen, type 2 is an alertDialog box
+    private void generateWarning(String message, int type) {
```

Comment:
Good idea :) You should consider putting it in a static Utility class, to be able to call it from elsewhere in the app. You should create an *Enum* for the type instead of using *int*. It would be cleaner and **far** clearer for a programmer that would read the code or use the method.